File: 19-meca_flu-eloudrhiri-jacquet-navez/deriv.m
function v = deriv(f_left, f_c, f_right, type_left, type_c, type_right,h)
%% COMPUTATION
if(type_left == 0 && type_c == 0 && type_right == 0)
   % trivial case
   v = 0;
elseif(type_left == 0) 
    % rear centered derivative
    v = (f_right - f_c)/h; 
elseif(type_right ==0) 
    % front centered derivative
    v = (f_c - f_left)/h;
elseif(type_c ==0 && type_r ~= 0 && type_l ~=0) 
    v=(f_c - f_left)/h;
else
    % centered derivative -> most common case
    v = (f_right - f_left)/(2*h);
end
end

File: deriv.py
def deriv(f_left, f_c, f_right, type_left, type_c, type_right, h):
    if type_left == 0 and type_c == 0 and type_right == 0:
        v = 0

    elif type_left == 0:
        v = (f_right - f_c) / h

    elif type_right == 0:
        v = (f_c - f_left) / h

    elif type_c == 0 and type_right != 0 and type_left != 0:
        v = (f_c - f_left) / h

    else:
        v = (f_right - f_left) / (2 * h)

    return v
